### PR TITLE
Dockerize sfn client #65

### LINKED ISSUE #65: Dockerize sfn-client
Implement docker setup for development

### PR DESCRIPTION
**Pull request by daumie**
### What does this PR do?
Provides an option to run the `sfn-client` using docker.

#### Description of Task to be completed?
- Create Dockerfile with instructions on how to create the `sfn-client` image.

#### How should this be manually tested?

1. On terminal, run command to create a docker image of the application
> docker build -f docker/Dockerfile -t sfn_client:production .

2. Run the created image with
> docker run -p 80:80 sfn_client:production

3. Access the application on:
> http://127.0.0.1:80

#### Any background context you want to provide?
Docker has several advantages i.e

- Docker produces an API for container management
- Containers are transferable across machines 
- Light and minimal overhead – Docker images are typically very small, which promotes rapid delivery and reduces the time to deploy new application containers.

Please Google for many more advantages of using Docker.

#### What are the relevant Github Issues ?

Fixes #65 

#### Screenshots (If applicable)
N/A

### CODE REVIEW

**Comment by daumie** [comment]
@75pollet , @SherSpock We can  review this  together before it is merged. What time is good for you?
cc @FedericoEsparza

**Comment by yakryder** [comment]
@daumie I would love that. I can do 15 minutes this afternoon if it isn't too late in the day for you -- weekend is freer.

No reason to hold this up though. 

I think most or all of what I would need from my own side to feel satisfied approving would be to build the image, spin up a container, and put it through the smokiest of manual smoke tests. The smoke test seems to me to be pretty much all or nothing -- if it builds, serves the container, and that container is reachable, I don't understand on a cursory analysis of the current architecture and scope of changes any way that we could thereby be breaking nominal or introducing deviant functionality.

**Comment by daumie** [comment]
Let's do weekend. Saturday as from **11:00 UTC** . Will this work fo you?
cc @75pollet , @FedericoEsparza ?

**Comment by yakryder** [comment]
I can do that. I'll be on!

File: docker/Dockerfile
Comment (yakryder) [inline docker/Dockerfile]: Dangling comment

File: docker-config/vhost.conf
Comment (yakryder) [inline docker-config/vhost.conf]: We talked about this on call -- you spotted this as a missing newline

File: docker/Dockerfile
Comment (yakryder) [inline docker/Dockerfile:15]: Very well-optimized from what I understand. 22 megabytes is a great image size compared to 600+

File: .dockerignore
Comment (yakryder) [inline .dockerignore:8]: Good exclusions

**Review by yakryder — APPROVED** [review]
I think this looks great! Disciplined and effective changeset, and happy to learn from you about Docker.

We could add some output when you serve the container that the app is running properly, now or in a future PR as you suggested. 

Few potential changes, but it builds and works so I wouldn't feel sad if this went in now

**Pull request merged**
